tests/test_instance.py

```diff
@@ -13,6 +14,34 @@
 from conftest import CREATE_INSTANCE_METHOD, DELETE_INSTANCE_METHOD, GET_INSTANCE_METHOD
 
 
+@pytest.fixture(scope="session", autouse=True, params=[True, False])
+def configuration_mock(request):
+    pytest.isSecure = request.param
+    # A valid configuration file setting up the uri and metadata
+    config_path = str("tests/config.json")
+    config = (
+        r"""{
+    "version": 1,
+    "pim": {
+        "uri": "dns:instancemanagement.example.com:443",
+        "headers": {
+            "authorization": "Bearer 007"
+        },
+        "tls": %s
+    }
+}"""
+        % str(request.param).lower()
+    )
+
+    with open(config_path, "w") as f:
+        f.write(config)
+
+    # Connect the client based on this configuration
+    # and run a request
+    os.environ["ANSYS_PLATFORM_INSTANCEMANAGEMENT_CONFIG"] = config_path
```

Comment:
Changing the environment in the test can be a bit treacherous as it can impacts more tests than the initial one. At the moment it does not matter most since the fixture has the `session` scope and in `autouse`, but if at some point this fixture is only applied to some tests, then it can cause some headaches. You can make it future proof by patching the environment for the lifetime of the fixture:

```py
with patch.dict(os.environ, {"ANSYS_PLATFORM_INSTANCEMANAGEMENT_CONFIG": config_path}):
    pytest.configuration = pypim.Configuration.from_environment()
    yield
```

The tests using this fixture are executed as part of the `yield`, and once they are terminated, the `patch` goes out of scope and the environment is left untouched